compiler/util/stringutil.cpp

```diff
@@ -435,16 +435,17 @@ std::string ltrimAllLines(std::string s) {
 }
 
 /*
- * Gather words from the string and store them into the array.
- * These words are arguments to a program.
+ * Split a string separated by a delimiter into a vector of substrings
```

Comment:
Hmm... this also splits on newlines since you used `getline` right? I don't see why that would or wouldn't be better. Is that intentional?

Often split functions either split on whitespace or on a particular character. What if it had an argument to split on all whitespace?

Reply:
Yes, to be able to split by newlines is intentional. I have also added a param to split on all whitespaces.